### PR TITLE
feat(translator): add support for configuring the GRPC Health Checker

### LINKED ISSUE #4044: Support gRPC Healthchecks
*Description*:
Envoy Gateway currently supports gRPC routes and integration with gRPC services (ext-auth, ext-proc, telemetry, ... ). gRPC services often support a standardized health check service, which is currently supported by Envoy. Envoy Gateway should extend active health check support to include gRPC health checks. 

*Relevant Links*:
* https://github.com/grpc/grpc/blob/master/doc/health-checking.md
* https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck-grpchealthcheck

### PR DESCRIPTION
**Pull request by liorokman**
**What this PR does / why we need it**:
This PR adds support GRPC health checks.

**Which issue(s) this PR fixes**:
Fixes #4044

### CODE REVIEW

**Comment by guydc** [comment]
@liorokman  - can we add support for configuring the grpc `service_name`? Similar to HTTP HC `path`, it could be necessary to support a custom HC service name for grpc. 

https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck-grpchealthcheck

**Review by guydc — APPROVED** [review]
LGTM, thanks!

**Review by arkodg — APPROVED** [review]
LGTM thanks !

File: api/v1alpha1/healthcheck_types.go [hunk 1]
```diff
@@ -159,6 +167,15 @@ type TCPActiveHealthChecker struct {
 	Receive *ActiveHealthCheckPayload `json:"receive,omitempty" yaml:"receive,omitempty"`
 }
 
+// GRPCActiveHealthChecker defines the settings of the GRPC health check.
+type GRPCActiveHealthChecker struct {
+	// ServiceName to send in the health check request.
+	// If this is not specified, then the health check request applies to the entire
+	// server and not to a specific service.
+	// +optional
+	ServiceName *string `json:"serviceName,omitempty" yaml:"serviceName,omitempty"`
```
Comment (arkodg) [inline api/v1alpha1/healthcheck_types.go:176]: probably merged this too soon, 
thoughts on renaming this to `Service` to keep it consistent with gateway api 
https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.GRPCMethodMatch

Comment (liorokman) [reply api/v1alpha1/healthcheck_types.go:176]: I used `ServiceName` to be consistent with the Envoy Proxy API: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck-grpchealthcheck 

I can open a new PR to change to `Service` if you want.

Comment (guydc) [reply api/v1alpha1/healthcheck_types.go:176]: +1 to being more consistent with GW-API

Comment (arkodg) [reply api/v1alpha1/healthcheck_types.go:176]: thanks @liorokman +1 to renaming to `Service`

**Pull request merged**